cachi2/core/package_managers/yarn/main.py

```diff
@@ -73,20 +76,28 @@ def _configure_yarn_version(project: Project) -> None:
             ),
         )
 
-    if (
-        yarn_path_version
-        and package_manager_version
-        and yarn_path_version != package_manager_version
-    ):
+    if yarn_path_version and package_manager_version:
```

Comment:
Do we need to nest an additional if/else here? Once the code gets passed the exception checks, either one of yarn_path_version or package_manager_version is defined or if they both are, they are equal.

Couldn't we just have `version = yarn_path_version if yarn_path_version else package_manager_version` right before we check whether the version is within the valid range?

Reply:
Right, moved the check to the top and dropped the extra nesting.